### PR TITLE
feat(compass-schema): update schema toolbar to leafygreen components COMPASS-5674

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-5674

This PR updates the schema tab's toolbar, we're creating a new `SchemaToolbar` component (in ts). Also updated a few files to typescript that didn't take many changes.

In the screenshots below the background is still gray. This will be changed when we update the collection view background to white in a future pr.

Screen Shot 2022-06-02 at 12 28 23 PM
Screen Shot 2022-06-02 at 12 39 07 PM

### CODE REVIEW

File: packages/compass-schema/src/components/schema-toolbar/schema-toolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    const queryBarRole = localAppRegistry.getRole('Query.QueryBar')![0];
+
+    queryBarRef.current = {
+      component: queryBarRole.component,
+      store: localAppRegistry.getStore(queryBarRole.storeName!),
+      actions: localAppRegistry.getAction(queryBarRole.actionName!),
+    };
+
+    setQueryBarLoaded(true);
+  }, [localAppRegistry]);
```
Comment (gribnoysup) [inline packages/compass-schema/src/components/schema-toolbar/schema-toolbar.tsx:119]: Why is this happening in the effect?

Comment (Anemy) [reply packages/compass-schema/src/components/schema-toolbar/schema-toolbar.tsx]: Thanks for catching. Didn't need to be, not sure why it was. Moved it out 👍

File: packages/compass-schema/src/components/schema-toolbar/schema-toolbar.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  it('does not render errors when the analysis state is not error', function () {
+    renderSchemaToolbar({
+      errorMessage: 'test error msg',
+    });
+
+    expect(
+      screen.queryByText(
+        'An error occurred during schema analysis: test error msg'
+      )
+    ).to.not.exist;
```
Comment (lerouxb) [inline packages/compass-schema/src/components/schema-toolbar/schema-toolbar.spec.tsx:113]: The thing about these assertions is that any typo in the text above would also cause it to not exist.

Comment (Anemy) [reply packages/compass-schema/src/components/schema-toolbar/schema-toolbar.spec.tsx]: Updated it to also test with a `data-testid` so it's not all dependent on the text, and updated it to use a shared variable for the text, I think that'll make it a bit more resilient, thanks for the suggestion. Unit tests like these tests are fairly brittle. I usually try to keep tests independent of variables used in the actual code so ideally break / need to be updated when we change some text or rendering. Additionally I like to keep tests as non-coupled as possible so teardown and changes are simpler.

I'm not sure I follow the other suggestion, what would be the action?

**Pull request merged**